mermaid_py/client.py

```diff
@@ -0,0 +1,443 @@
+
+    def get_obs_data(self, obs, filter, filter_val=None, name=None, id=None):
+        """
+        Gets observation resources which are the lowest level of MERMAID data, representing individual observations
+        in sample unit methods, belonging to sample events (a set of sample unit methods at a site on a specific date).
+        See https://mermaid-api.readthedocs.io/en/latest/projects.html#observations for more information on
+        observations.
+        :param obs: Project observations include; obstransectbeltfishs, obsbenthiclits, obsbenthicpits,
+        obshabitatcomplexities, obscoloniesbleached, obsquadratbenthicpercent.
+        :type obs: str
+        :param filter: Visit https://mermaid-api.readthedocs.io/en/latest/projects.html#observations for list of
+        observations filter options.
+        :type filter: str
+        :param filter_val: (optional) Required for filters requiring values eg. size_min, size_max, count_min,
+        count_max, length_min, length_max filters.
+        :type filter_val: int
+        :param name: (optional if 'id' provided) MERMAID project name.
+        :type name: str
+        :param id: (optional if 'name' provided) MERMAID project ID.
+        :type id: str
+        :return: observation data.
+        :rtype: dict
+        """
+        obs_filters = {
+            "obstransectbeltfishs": [
+                "beltfish",
+                "beltfish__transect",
+                "beltfish__transect__sample_event",
+                "fish_attribute",
+                "size_min",
+                "size_max",
+                "count_min",
+                "count_max",
+            ],
+            "obsbenthiclits": [
+                "benthiclit",
+                "benthicpit__transect",
+                "benthiclit__transect__sample_event",
+                "attribute",
+                "growth_form",
+                "length_min",
+                "length_max",
+            ],
+            "obsbenthicpits": [
+                "benthicpit",
+                "benthicpit__transect",
+                "benthicpit__transect__sample_event",
+                "attribute",
+                "growth_form",
+            ],
+            "obshabitatcomplexities": [
+                "habitatcomplexity",
+                "habitatcomplexity__transect",
+                "habitatcomplexity__transect__sample_event",
+                "score",
+            ],
+        }
+        # Valid observation and filter for observation.
+        if obs not in obs_filters:
+            raise Exception(f"Invalid observation:{obs}")
+        if filter not in obs_filters[obs]:
+            raise Exception(f"Invalid observation filter:{filter}")
+
+        # API observations path.
+        return self._api_projects_path(
+            name=name, id=id, resource=obs, filter=filter, filter_val=filter_val
+        )
+
+    def get_sample_unit(self, unit, filter=None, filter_val=None, name=None, id=None):
```

Comment:
This and `get_sample_method` below should both be plural, right? Matching the method for sample events?